vignettes/nactdt.Rmd

```diff
@@ -0,0 +1,194 @@
+)
+```
+
+## Single Source{#input}
+
+To derive the New Anti-Cancer Therapy start date with data from a single source, the function `admiral::derive_vars_merged()` can be used. If numeric date is required, pre-processing using `admiral::derive_vars_dt()` can be used, or if time part is needed, `admiral::derive_vars_dt()` can be used.
+
+```{r message=FALSE}
+cm_ext <- derive_vars_dt(
+  cm,
+  dtc = CMSTDTC,
+  new_vars_prefix = "CMST"
+)
+
+adsl <- derive_vars_merged(
+  admiral_adsl,
+  dataset_add = cm_ext,
+  by_vars = exprs(STUDYID, USUBJID),
+  order = exprs(CMSTDT),
+  mode = "first",
+  new_vars = exprs(NACTDT = CMSTDT),
+  filter_add = CMSCAT == "CHEMOTHERAPY" & CMCAT == "ON TREATMENT"
+)
+```
+
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(
+  adsl,
+  display_vars = exprs(USUBJID, NACTDT),
+  filter = !is.na(NACTDT)
+)
+```
+
+## Multiple Sources - Prerequisite Steps {#multiinput}
+
+To derive the New Anti-Cancer Therapy start date with data from multiple sources, the function `admiral::derive_var_extreme_dt()` can be used, or if time part is needed, `admiral::derive_var_extreme_dtm()` can be used.
+ 
+ 
+### Steps 
+
+Prior to calling the function, two steps must be performed. To start, an input `date_source` object must be created for each source dataset using `admiral::date_source()`. Within this function, any company-specific subsetting can be passed to the `filter` argument. 
+
+```{r message=FALSE}
+cm_date <- date_source(
+  dataset_name = "cm",
+  filter = CMSCAT == "CHEMOTHERAPY" & CMCAT == "ON TREATMENT" & !is.na(CMSTDT),
+  date = CMSTDT
+)
+
+pr_date <- date_source(
+  dataset_name = "pr",
+  filter = PRCAT == "CANCER RELATED" & PRSCAT == "ON TREATMENT" & !is.na(PRSTDT),
+  date = PRSTDT
+)
+```
+
+Lastly, numeric version of the dates will be needed, conversion can be done using `admiral::derive_vars_dt()`, or if time part is needed, `admiral::derive_vars_dtm()`. 
+
+
+```{r message=FALSE}
+cm_ext <- cm %>%
+  derive_vars_dt(
+    dtc = CMSTDTC,
+    new_vars_prefix = "CMST"
+  )
+
+pr_ext <- pr %>%
+  derive_vars_dt(
+    dtc = PRSTDTC,
+    new_vars_prefix = "PRST"
+  )
+```
+
+## Multiple Sources - Derive New-Anti Cancer Start Date and Merge with `ADSL` {#nactdt}
+
+After completion of the prerequisite steps, the new anti-cancer date can be derived while simultaneously adding it to `ADSL`.
+
+```{r message=FALSE}
+adsl <- admiral_adsl %>%
+  derive_var_extreme_dt(
+    new_var = NACTDT,
+    cm_date, pr_date,
+    source_datasets = list(
+      cm = cm_ext,
+      pr = pr_ext
+    ),
+    mode = "first"
+  )
+```
+
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(
+  adsl,
+  display_vars = exprs(USUBJID, NACTDT),
+  filter = !is.na(NACTDT)
+)
+```
+## Derive Date Parameter {#par}
+
+For deriving parameter for new anti-cancer date from a single source, `admiral::derive_param_extreme_event()` can be used, please consult the `{admiralonco}` [Creating ADRS](adrs.html) for guidance on creating parameters with examples.
```

Comment:
Can we provide the text or example of creating the record with a new parameter. And also, we we can mentioned, that the record can be added to ADEVENT or ADDATES as per company/ study specific intermediate dataset for ADTTE.
Also, can we also show the use case of this NACDT date. We only need the link to ADTTE vignette to section "Creating Your Own Time-to-Event Source Objects" where we have shown the example.

Reply:
@vinhn23 did you added text as mentioned above? @bundfussr please let us know what you think if we need to add ADTTE reference. Also, mention in text for ADVENT,ADDATES

Reply:
@amitjaingsk  This section was updated:

## Derive Date Parameter {#par}

For deriving parameter for new anti-cancer date from a single source, `admiral::derive_param_extreme_event()` may be used, please consult the `{admiralonco}` [Creating ADRS](adrs.html) for guidance on creating parameters and on the usage of `NACTDT` variable with examples. Additionally, subsection *Creating Your Own Time-to-Event Source Objects* in the `{admiralonco}` [Creating ADTTE](adtte.html) vignette may be referenced for case usage.


As of `{admiral}` version 0.11.0, `admiral::derive_param_extreme_record()` is recommended to be used for deriving parameter based on the first or last observation from single and multiple sources. 

Based on individual company standards, this parameter may be added to an event or date dataset such as `ADEVENT` or `ADDATES`, which are generally intermediate datasets to `ADTTE`. For demonstration purpose, a new anti-cancer date parameter will be merged into `ADRS`. A list of `record_source()` objects is expected and this will contain the sources of the dates and any company specific filtering.

Reply:
I wonder if this is the right place for referring to ADRS and ADTTE, especially as in the examples in the other vignettes it is assumed that the new anti-cancer start date is provided as variable.